src/test/compile-fail/rfc-2008-non-exhaustive/unit_struct_match_exhaustively.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright 2012 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// gate-test-non_exhaustive
+#![feature(non_exhaustive)]
+
+// aux-build:structs.rs
+extern crate structs;
+
+use structs::UnitStruct;
+
+// This errors as we can't exhaustively (without the ..) match
+// a struct from a external crate marked as non-exhaustive.
+
+fn match_struct(us: &UnitStruct) {
+    let &UnitStruct { } = us;
+    //~^ ERROR `..` required with struct marked as non-exhaustive
```

Comment:
You can merge several test files testing similar things into one file with several `//~ ERROR` markers, it's usually more convenient that way (both for writing better more exhaustive tests, and for reviewing them).

Reply:
Changed this. It's a lot easier to keep track of now.